### ISSUE TITLE
Numeric ids and bibtex output

### ISSUE THREAD

**Issue by AlexanderZeilmann** [citation-js/citation-js#137]
I want to report a collection of bugs related to ids and bibtex output.

First of all, the following code works as intended
```
Cite({
    type: "book",
    id: "someID"
}).format('bibtex')
```

However, if the `id` is a string that can be parsed as number
```
Cite({
    type: "book",
    id: "1234"
}).format('bibtex')
```
we get the error
```
TypeError: id.replace is not a function
```

This is due to the following two things:

### 1. The `id` is parsed as a number.
Happening here:
https://github.com/citation-js/citation-js/blob/3f3eee0813c7d578a454c34e402fa342d0693cfa/packages/core/src/plugins/input/csl.js#L327
I think the line above is missing a `&& !fieldType.includes('string')`

The parsing also has various side effects, that lead to information loss:
* The `id`s `"0123"` and `"123"` are both mapped to `123`.
* The `id` `"123456789101112131415"` is mapped to `123456789101112130000`.

### 2. Calling the `String.prototype.replace` function on the parsed number
https://github.com/citation-js/citation-js/blob/0c690e6ab0f092e64e80ffc7bf2fc81ec580fcb9/packages/plugin-bibtex/src/mapping/shared.js#L279
As numbers are also valid `id`s we need to convert it to a string first:
`id.toString().replace(unsafeChars, '')`

There is also another minor bug on that line:
If the id is equal to the number `0`, then the line evaluates to `"undefined"` but should evaluate to "0".

I am happy to submit a pull request if my analysis of the bug is correct.

**Comment by larsgw** [comment]
Thank you for the detailed report! I will look into it.

> I think the line above is missing a `&& !fieldType.includes('string')`

The idea was that a number is generally better for most purposes but this is probably a good idea.

> I am happy to submit a pull request if my analysis of the bug is correct.

That would be great!

**Comment by AlexanderZeilmann** [comment]
Closed via #139

**Comment by larsgw** [comment]
I still have to release a patch version with the patch included but that should happen soon.
